src/uct/api/uct.h

```diff
@@ -1114,6 +1114,8 @@ struct uct_ep_params {
     /**
      * Connection request that was passed to
      * @ref uct_cm_listener_conn_request_args_t::conn_request .
+     * @note This handle will be released regardless of @ref uct_ep_create
+     * return status.
```

Comment:
well in the case of tcp_sockcm, this conn_request is the server's endpoint so it's not really released (until the end of the connection)
https://github.com/openucx/ucx/blob/master/src/uct/tcp/tcp_sockcm_ep.c#L724

Reply:
but if uct_ep_create(conn_request) fails, connection can't be established from user's point of view. so need to clean all related resources in UCT.

Reply:
right but if it succeeds, then conn_request won't be release. 
in rdmacm it will be released in uct_ep_create in any case.

Reply:
it's internal implementation, from user's perspective the request is released and can't be accessed.

Reply:
it can't be accessed but it's not released. why document that it's released if it's not?
sounds like releasing it in rdmacm is an implementation detail...

Reply:
if it can't be accesses or used, that means it's released.

Reply:
@alinask where will this request be released in tcp_sockcm in this case?

Reply:
BTW we can write it as something like
`@note After a call to this function, @a params.conn_request is consumed and should not be used anymore, even if the call returns with error `

Reply:
> @alinask where will this request be released in tcp_sockcm in this case?

@yosefe since this is the tcp_ep it will be released when the ep is destroyed. it's true that it should not be accessed by the user anymore in the form of conn_request and that we should say imo.

Reply:
> BTW we can write it as something like
> `@note After a call to this function, @a params.conn_request is consumed and should not be used anymore, even if the call returns with error `

i agree. this function - should specify uct_ep_create since this is the doc of ep_params.

Reply:
> @yosefe since this is the tcp_ep it will be released when the ep is destroyed.

When which ep is destroyed? If server ep create fails, then it's not created..

Reply:
the server creates a tcp_ep before the call to uct_ep_create, this ep is passed to the server's connect cb as the conn_request and then used in the uct_ep_create function
1. https://github.com/openucx/ucx/blob/master/src/uct/tcp/tcp_sockcm_ep.c#L431
2. https://github.com/openucx/ucx/blob/master/src/ucp/wireup/wireup_cm.c#L949
3. https://github.com/openucx/ucx/blob/master/src/uct/tcp/tcp_sockcm_ep.c#L753

currently in case of an error, the internal tcp_ep isn't released, i'll fix it in my next pr.